### PR TITLE
TN PF-261 Flight Debug Info

### PR DESCRIPTION
**Pull request by tnadler**
This PR contains the changes to:
1) Create a FlightDebugInfo class and pass it to the flight constructor.
2) Store that class as a JSON in the flight table (includes liquibase config to add a column to the table).
3) Logic to implement RestartEachStep - a way of running a flight where the flight stops and is resumed from the database at each step.

### CODE REVIEW

File: src/main/java/bio/terra/stairway/FlightDebugInfo.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+/**
+ * Debug information for a flight. Parameters here change how flights run to ensure debugability/
+ * testability. Designed to be easily extensible,
+ */
+public class FlightDebugInfo {
+
+  private static ObjectMapper objectMapper;
```
Comment (ddietterich) [inline src/main/java/bio/terra/stairway/FlightDebugInfo.java]: these can be `final` I think

File: src/test/java/bio/terra/stairway/DebugInfoTest.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The debug info test needs to ensure that the flight actually pauses between steps so it can be
+ * restarted. To do this we use a two step Flight. We start the flight, and then ensure after the
+ * initial submission the flight is left in a ready state and can be resumed. NOTE: This only works
+ * for the initial submit currently as we have not persisted FlightDebugInfo for reading from the
```
Comment (ddietterich) [inline src/test/java/bio/terra/stairway/DebugInfoTest.java]: Well, now you have! But the flight won't resume yet.

**Review by ddietterich — COMMENTED** [review]
A few nit comments. Otherwise, looks good, except for the constructor thing...

File: src/test/java/bio/terra/stairway/DebugInfoTest.java (hunk 2 above)
Comment (tnadler) [reply src/test/java/bio/terra/stairway/DebugInfoTest.java]: This file did not get updated somehow! Replaced all this text.!

File: src/main/java/bio/terra/stairway/FlightDebugInfo.java (hunk 1 above)
Comment (tnadler) [reply src/main/java/bio/terra/stairway/FlightDebugInfo.java]: Initializing object mapper this way looks odd to me somehow...
The restartEachStep can't (or at least I think can't) as to get Jackson to work I created a setter.

**Review by ddietterich — APPROVED** [review]
LGTM - thank you!

**Pull request merged**